varianteval/scripts/build_igv_batch_script.py

```diff
@@ -0,0 +1,128 @@
+                                     colorby_ins_size=colorby_ins_size,
+                                     groupby_pair_orientation=groupby_pair_orientation,
+                                     viewaspairs=viewaspairs)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Batch script generation')
+    parser.add_argument('--input_vcf', help='Input vcf file')
+    parser.add_argument('--bam_path', help='Path to BAM to be given to IGV')
+    parser.add_argument('--output_dir', help='Output directory where support vector sub-directories will be generated')
+    parser.add_argument('--trg_supp_vec', help='Target support vector to define subset of records that '
+                                               'will be included in batch script (NA to be used in '
+                                               'cases when the records dont have a SUPP_VEC field)')
+    parser.add_argument('--igv_screenshot_dir', help='Screenshot directory to be specified in batch script - useful'
+                                                     'when iterating over multiple sets of screenshots')
+    parser.add_argument('--min_svlen', type=int, default=0, help='Minimum sv length to include in batch script')
+    parser.add_argument('--max_svlen', type=int, default=250000, help='Maximum sv length to include in batch script')
+    parser.add_argument('--ref_genome', type=str, default='hg38', help='Reference genome to load into IGV')
+    parser.add_argument('--mode', choices=['non_survivor', 'survivor_single', 'survivor_multiple'],
+                        default='non_survivor',
+                        help='Different modes for generating a batch script: non_survivor - input VCF doesnt include'
+                             'SUPP_VEC field and all records will be added to the batch script; survivor_single - input'
+                             'VCF includes SUPP_VEC field and only the records with SUPP_VEC==args.trg_supp_vec will'
+                             'be added to the batch script; survivor_multiple - input VCF includes SUPP_VEC field and'
+                             'the screenshots from the records with the different SUPP_VEC values will be placed in'
+                             'directories named with the corresponding SUPP_VEC binary string')
+    # flags for different IGV visualization settings
+    parser.add_argument('--colorby_insert_size', type=bool, default=True, help='Adds IGV colorby INSERT SIZE')
+    parser.add_argument('--groupby_pair_orientation', type=bool, default=True, help='Adds groupby PAIR ORIENTATION')
+    parser.add_argument('--viewaspairs', type=bool, default=True, help='Adds viewaspairs')
+    args = parser.parse_args()
+
+    if args.igv_screenshot_dir is None:
+        igv_screenshot_dir = args.output_dir
+    else:
+        igv_screenshot_dir = args.igv_screenshot_dir
+
+    if args.mode == 'non_survivor':
```

Comment:
looks like the mode parameter is redundant with support vector -- the support vector param can be optional and if provided, should be used as an additional filter (similar to len or type)